lib/ansible/modules/network/asa/asa_og.py

```diff
@@ -0,0 +1,792 @@
+from __future__ import absolute_import, division, print_function
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: asa_og
+version_added: "2.8"
+author:
+    - "Federico Olivieri (@Federico87)"
+short_description: Manage object groups on a Cisco ASA
+description:
+    - This module allows you to create and update object-group network/service on Cisco ASA device.
+extends_documentation_fragment: asa
+options:
+    name:
+        description:
+            - object-group name
+        required: true
+    group_type:
+        description:
+            - object-group type
+        choices: ['network-object', 'service-object', 'port-object']
+        required: true
+    protocol:
+        description:
+            - protocol for object-group service with port-object
+        choices: ['udp', 'tcp', 'tcp-udp']
+    host_ip:
+        description:
+            - host ip address for object-group network i.e. "network-object host 8.8.8.8"
+        type: list
+    description:
+        description:
+            - object-group description
+    group_object:
+        description:
+            - group-object for network object-group i.e. "network-object object-group ansible_test"
+        type: list
+    ip_mask:
+        description:
+            - IP address and mask for network object-group i.e. "network-object 192.168.0.0 255.255.0.0"
+        type: list
+    port_range:
+        description:
+            - port range for port-object i.e. "port-object range 0 1024"
+    port_eq:
+        description:
+            - single port for port-object i.e. "port-object eq www"
+    service_cfg:
```

Comment:
why can't we introduce a argument with field as: `tcp_destination_eq` instead of `service_cfg`, where user can give input as `8080/www` and it'll be easier for user as well.
```suggestion
    tcp_destination_eq: 8080
```

Reply:
@justjais  For service object there are many combinations, not always is tcp/destiantion/eq combination, so it would mean to have a key in the module that change every time. Also, and probably,more relevant, in some cases port is not required. 
For example:

```
object-group service ansible_test
 service-object tcp source range 0 65535
 service-object udp destination eq domain
 service-object tcp destination lt 1024
 service-object eigrp
 service-object tcp-udp source neq www
 service-object pim
 service-object ip
```

Reply:
@FedericoOlivieri OK, you can pass on the service-object as `list` as done earlier.